test/emu_mps/test_endtoend.py

```diff
@@ -231,33 +310,36 @@ def test_end_to_end_afm_ring() -> None:
         t_fall=t_fall,
     )
 
-    result = simulate(seq)
-
-    bitstrings = result.bitstrings[-1]
-    final_state = result.state[-1]
-    final_fidelity = result.fidelity_1[-1]
-    max_bond_dim = final_state.get_max_bond_dim()
-    fidelity_state = create_antiferromagnetic_mps(num_qubits)
-
-    assert bitstrings["1010101010"] == 129  # -> fidelity as samples increase
-    assert bitstrings["0101010101"] == 135
-    assert fidelity_state.overlap(final_state) == approx(final_fidelity, abs=1e-10)
+    result = simulate(seq, optimize_qubit_ordering=optimize_order)
+
+    final_time = -1
+    bitstrings = result.bitstrings[final_time]
+    state_fin = result.state[final_time]
+    fidelity_fin = result.fidelity_1[final_time]
+    max_bond_dim = state_fin.get_max_bond_dim()
+    fidelity_st = create_antiferromagnetic_mps(num_qubits)
+    occupation = result.occupation[final_time]
+    energy = result.energy[final_time]
+    energy_variance = result.energy_variance[final_time]
+    second_moment_energy = result.energy_second_moment[final_time]
+
+    assert bitstrings["1010101010"] == bs_count[0]
+    assert bitstrings["0101010101"] == bs_count[1]
+    assert fidelity_st.overlap(state_fin) == approx(fidelity_fin, abs=1e-10)
     assert max_bond_dim == 29
 
-    q_density = result.occupation[-1]
-    assert approx(q_density, 1e-3) == [0.578] * 10
-
-    energy = result.energy[-1]
-    assert approx(energy, 1e-8) == -115.34370829396005
+    # Comparing against EMU-SV -- state vector emulator
+    assert approx(occupation, abs=1e-3) == [0.5782] * 10
+    assert approx(occupation, rel=1e-3) == [0.5782] * 10
 
-    energy_variance = result.energy_variance[-1]
-    assert approx(energy_variance, 1e-6) == 45.905980469959104
+    assert approx(energy, abs=1e-2) == -115.34554479213088
+    assert approx(energy, rel=1e-4) == -115.34554479213088
 
-    second_moment_energy = result.energy_second_moment[-1]
-    assert approx(second_moment_energy, 1e-6) == 13350.07680148
+    assert approx(energy_variance, abs=1e-1) == 45.91111056399
```

Comment:
mmh abs=1e-1 and 45.91111056399
`45.9` would be good

Reply:
Instead of modifying this existing test by changing tolernaces, what about creating (a) new test(s) that checks that observables are identical/similar between with and without optimatrix? That lets us also see what is the difference to expect. So simulate twice same sequence with and without optimatrix

Reply:
previous tests were benchmarking emu-mps against emu-mps itself. We take a run and consider its results as correct. Now I do it against emu-sv. I compare here permuted and unpermuted emu-mps against emu-sv and show that quality didn't drop with permutation.

Reply:
is it an idea to use emu-sv directly instead of hardcoding values like `45.91111056399`? With new API it should be really easy to implement

Reply:
I'm a bit hesitant here. The idea is good, I'm afraid it implementation will require changes in other places. Propose to do that separately. For me it sounds like uniformising and generalising the tests